packages/grid/_modules_/grid/utils/EventManager.ts

```diff
@@ -38,30 +60,36 @@ export class EventManager {
   }
 
   removeListener(eventName: string, listener: EventListener): void {
-    if (Array.isArray(this.events[eventName])) {
-      const idx = this.events[eventName].indexOf(listener);
-
-      if (idx > -1) {
-        this.events[eventName].splice(idx, 1);
-      }
+    if (this.events[eventName]) {
+      this.events[eventName].regular.delete(listener);
+      this.events[eventName].highPriority.delete(listener);
     }
   }
 
-  removeAllListeners(eventName?: string): void {
-    if (!eventName) {
-      this.events = {};
-    } else if (Array.isArray(this.events[eventName])) {
-      this.events[eventName] = [];
-    }
+  removeAllListeners(): void {
+    this.events = {};
   }
 
   emit(eventName: string, ...args: any[]): void {
-    if (Array.isArray(this.events[eventName])) {
-      const listeners = this.events[eventName].slice();
-      const length = listeners.length;
+    const collection = this.events[eventName];
+    if (!collection) {
+      return;
+    }
+
+    const highPriorityListeners = Array.from(collection.highPriority.keys());
+    const regularListeners = Array.from(collection.regular.keys());
+
+    for (let i = highPriorityListeners.length - 1; i >= 0; i -= 1) {
+      const listener = highPriorityListeners[i];
+      if (collection.highPriority.has(listener)) {
```

Comment:
Could you add a test resuming what you said in https://github.com/mui-org/material-ui-x/pull/3206#issuecomment-976678291?

Besides the separation of the listeners in two groups, the biggest value of this PR on this line (and the equivalent for regular events).

Reply:
Yes, this PR was supposed to be a one liner if I did not have to change to a Map.
I'll add a few tests on the `EventManager` :+1:

Reply:
Done